Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-77.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: 58491e8c20ff19b22d491f9623b5fe14885fa129
Containerized:False
Number of Error(s): 5
src-react/components/LogicalFileSummary.tsx(161,107): error TS2339: Property AccessCost does not exist on type LogicalFile.
src-react/components/LogicalFileSummary.tsx(162,107): error TS2339: Property AtRestCost does not exist on type LogicalFile.
make[2]: *** [esp/src/CMakeFiles/eclwatch.dir/build.make:330: esp/src/build/dist/stub.eclwatch.js] Error 1
make[1]: *** [CMakeFiles/Makefile2:12240: esp/src/CMakeFiles/eclwatch.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2022-04-13_17-41-54
time logs started
Prep time   :    23 sec (00:00:23)
Build time  :   167 sec (00:02:47)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   190 sec (00:03:10)
time logs end

[Build error:](https://gist.github.com/5eb59899acc84e5d07451be734d8040d)
Build log [Build log](https://gist.github.com/HPCCSmoketest/5eb59899acc84e5d07451be734d8040d#file-result-22-04-13-17-38-26-log)

--- a/esp/src/eclwatch/DFUQueryWidget.js
+++ b/esp/src/eclwatch/DFUQueryWidget.js
@@ -27,6 +27,7 @@ define([
     "hpcc/DelayLoadWidget",
     "src/WsTopology",
     "src/Utility",
+    "src/Session",
 
     "put-selector/put",
 
@@ -61,7 +62,7 @@ define([
 ], function (declare, lang, nlsHPCCMod, arrayUtil, domClass, domForm, topic,
     registry, Dialog, Menu, MenuItem, MenuSeparator, PopupMenuItem,
     editor, selector, tree,
-    _TabContainerWidget, WsDfu, FileSpray, ESPUtil, ESPLogicalFile, ESPDFUWorkunit, DelayLoadWidget, WsTopology, Utility,
+    _TabContainerWidget, WsDfu, FileSpray, ESPUtil, ESPLogicalFile, ESPDFUWorkunit, DelayLoadWidget, WsTopology, Utility, Session,
     put,
     template) {
 
@@ -673,7 +674,31 @@ define([
                             node.innerText = Utility.valueCleanUp(value);
                         },
                     },
-                    Modified: { label: this.i18n.ModifiedUTCGMT, width: 162 }
+                    MinSkew: {
+                        label: nlsHPCC.MinSkew, width: 60,
+                        formatter: function (value, row) {
+                            return value !== undefined ? value : "";
+                        }
+                    },
+                    MaxSkew: {
+                        label: nlsHPCC.MaxSkew, width: 60,
+                        formatter: function (value, row) {
+                            return value !== undefined ? value : "";
+                        }
+                    },
+                    Modified: { label: this.i18n.ModifiedUTCGMT, width: 162 },
+                    AtRestCost: {
+                        label: nlsHPCC.FileCostAtRest, width: 100,
+                        formatter: function (cost, row) {
+                            return Session.formatCost(cost);
+                        }
+                    },
+                    AccessCost: {
+                        label: nlsHPCC.FileAccessCost, width: 100,
+                        formatter: function (cost, row) {
+                            return Session.formatCost(cost);
+                        }
+                    }
                 }
             }, this.id + "WorkunitsGrid");
             this.workunitsGrid.on(".dgrid-row-url:click", function (evt) {

--- a/esp/src/eclwatch/LFDetailsWidget.js
+++ b/esp/src/eclwatch/LFDetailsWidget.js
@@ -461,7 +461,7 @@ define([
             } else if (name === "RecordSize" && newValue === "0") {
                 this.updateInput("RecordSize", oldValue, this.i18n.NoPublishedSize);
             } else if (name === "Cost") {
-                this.updateInput("FormattedCost", oldValue, Session.formatCost(newValue || 0) + " (" + dojoConfig.currencyCode + ")");
+                this.updateInput("FormattedCost", oldValue, Session.formatCost(newValue));
             }
         },
 

--- a/esp/src/eclwatch/WUDetailsWidget.js
+++ b/esp/src/eclwatch/WUDetailsWidget.js
@@ -609,9 +609,9 @@ define([
                 var domElem = registry.byId(this.id + "ServiceNamesCustom");
                 domElem.set("value", newValue.Item.join("\n"));
             } else if (name === "ExecuteCost") {
-                this.updateInput("FormattedExecuteCost", oldValue, Session.formatCost(newValue || 0) + " (" + dojoConfig.currencyCode + ")");
+                this.updateInput("FormattedExecuteCost", oldValue, Session.formatCost(newValue));
             } else if (name === "FileAccessCost") {
-                this.updateInput("FormattedFileAccessCost", oldValue, Session.formatCost(newValue || 0) + " (" + dojoConfig.currencyCode + ")");
+                this.updateInput("FormattedFileAccessCost", oldValue, Session.formatCost(newValue));
             }
             if (name === "__hpcc_changedCount" && newValue > 0) {
                 var getInt = function (item) {

--- a/esp/src/package.json
+++ b/esp/src/package.json
@@ -43,7 +43,7 @@
     "@hpcc-js/chart": "2.76.1",
     "@hpcc-js/codemirror": "2.59.1",
     "@hpcc-js/common": "2.66.1",
-    "@hpcc-js/comms": "2.73.5",
+    "@hpcc-js/comms": "2.75.0",
     "@hpcc-js/dataflow": "7.0.0",
     "@hpcc-js/eclwatch": "2.72.8",
     "@hpcc-js/graph": "2.79.4",

--- a/esp/src/src-react/components/FileDetails.tsx
+++ b/esp/src/src-react/components/FileDetails.tsx
@@ -1,5 +1,6 @@
 import * as React from "react";
 import { Pivot, PivotItem } from "@fluentui/react";
+import { DFUDefFileFormat } from "@hpcc-js/comms";
 import { SizeMe } from "react-sizeme";
 import nlsHPCC from "src/nlsHPCC";
 import { FileParts } from "./FileParts";
@@ -39,8 +40,8 @@ export const FileDetails: React.FunctionComponent<FileDetailsProps> = ({
             replaceUrl(`/files/${file.NodeGroup}/${logicalFile}`);
         }
     }, [cluster, file?.NodeGroup, logicalFile]);
-    const [defFile] = useDefFile(cluster, logicalFile, "def");
-    const [xmlFile] = useDefFile(cluster, logicalFile, "xml");
+    const [defFile] = useDefFile(cluster, logicalFile, DFUDefFileFormat.Def);
+    const [xmlFile] = useDefFile(cluster, logicalFile, DFUDefFileFormat.Xml);
 
     const isDFUWorkunit = file?.Wuid?.length && file?.Wuid[0] === "D";
 

--- a/esp/src/src-react/components/Files.tsx
+++ b/esp/src/src-react/components/Files.tsx
@@ -8,7 +8,6 @@ import * as Utility from "src/Utility";
 import nlsHPCC from "src/nlsHPCC";
 import { useConfirm } from "../hooks/confirm";
 import { useFluentPagedGrid } from "../hooks/grid";
-import { useBuildInfo } from "../hooks/platform";
 import { HolyGrail } from "../layouts/HolyGrail";
 import { pushParams } from "../util/history";
 import { AddToSuperfile } from "./forms/AddToSuperfile";
@@ -77,7 +76,6 @@ export const Files: React.FunctionComponent<FilesProps> = ({
     const [showDesprayFile, setShowDesprayFile] = React.useState(false);
     const [mine, setMine] = React.useState(false);
     const [uiState, setUIState] = React.useState({ ...defaultUIState });
-    const [, { currencyCode }] = useBuildInfo();
 
     //  Grid ---
     const gridStore = React.useMemo(() => {
@@ -162,17 +160,23 @@ export const Files: React.FunctionComponent<FilesProps> = ({
                     node.innerText = Utility.valueCleanUp(value);
                 },
             },
+            MinSkew: {
+                label: nlsHPCC.MinSkew, width: 60, formatter: (value, row) => value ?? ""
+            },
+            MaxSkew: {
+                label: nlsHPCC.MaxSkew, width: 60, formatter: (value, row) => value ?? ""
+            },
             Modified: { label: nlsHPCC.ModifiedUTCGMT, width: 162 },
             AtRestCost: {
                 label: nlsHPCC.FileCostAtRest, width: 100,
                 formatter: function (cost, row) {
-                    return `${formatCost(cost ?? 0)} (${currencyCode || "$"})`;
+                    return `${formatCost(cost)}`;
                 }
             },
             AccessCost: {
                 label: nlsHPCC.FileAccessCost, width: 100,
                 formatter: function (cost, row) {
-                    return `${formatCost(cost ?? 0)} (${currencyCode || "$"})`;
+                    return `${formatCost(cost)}`;
                 }
             }
         }

--- a/esp/src/src-react/components/LogicalFileSummary.tsx
+++ b/esp/src/src-react/components/LogicalFileSummary.tsx
@@ -1,5 +1,6 @@
 import * as React from "react";
 import { CommandBar, ContextualMenuItemType, ICommandBarItemProps, ScrollablePane, ScrollbarVisibility, Sticky, StickyPositionType } from "@fluentui/react";
+import { DFUChangeProtection, DFUChangeRestriction } from "@hpcc-js/comms";
 import nlsHPCC from "src/nlsHPCC";
 import { formatCost } from "src/Session";
 import * as WsDfu from "src/WsDfu";
@@ -7,7 +8,6 @@ import * as Utility from "src/Utility";
 import { getStateImageName, IFile } from "src/ESPLogicalFile";
 import { useConfirm } from "../hooks/confirm";
 import { useFile } from "../hooks/file";
-import { useBuildInfo } from "../hooks/platform";
 import { ShortVerticalDivider } from "./Common";
 import { TableGroup } from "./forms/Groups";
 import { CopyFile } from "./forms/CopyFile";
@@ -32,7 +32,6 @@ export const LogicalFileSummary: React.FunctionComponent<LogicalFileSummaryProps
 
     const [file, isProtected, , refresh] = useFile(cluster, logicalFile);
     const [description, setDescription] = React.useState("");
-    const [, { currencyCode }] = useBuildInfo();
     const [_protected, setProtected] = React.useState(false);
     const [restricted, setRestricted] = React.useState(false);
     const [canReplicateFlag, setCanReplicateFlag] = React.useState(false);
@@ -105,8 +104,8 @@ export const LogicalFileSummary: React.FunctionComponent<LogicalFileSummaryProps
                 file?.update({
                     UpdateDescription: true,
                     FileDesc: description,
-                    Protect: _protected ? "1" : "2",
-                    Restrict: restricted ? "1" : "2",
+                    Protect: _protected ? DFUChangeProtection.Protect : DFUChangeProtection.Unprotect,
+                    Restrict: restricted ? DFUChangeRestriction.Restrict : DFUChangeRestriction.Unrestricted,
                 });
             }
         },
@@ -160,7 +159,8 @@ export const LogicalFileSummary: React.FunctionComponent<LogicalFileSummaryProps
                 "NodeGroup": { label: nlsHPCC.ClusterName, type: "string", value: file?.NodeGroup, readonly: true },
                 "Description": { label: nlsHPCC.Description, type: "string", value: description },
                 "JobName": { label: nlsHPCC.JobName, type: "string", value: file?.JobName, readonly: true },
-                "Cost": { label: nlsHPCC.Cost, type: "string", value: `${formatCost(file?.Cost ?? 0)} (${currencyCode})`, readonly: true },
+                "AccessCost": { label: nlsHPCC.FileAccessCost, type: "string", value: `${formatCost(file?.AccessCost)}`, readonly: true },
+                "AtRestCost": { label: nlsHPCC.FileCostAtRest, type: "string", value: `${formatCost(file?.AtRestCost)}`, readonly: true },
                 "isProtected": { label: nlsHPCC.Protected, type: "checkbox", value: _protected },
                 "isRestricted": { label: nlsHPCC.Restricted, type: "checkbox", value: restricted },
                 "ContentType": { label: nlsHPCC.ContentType, type: "string", value: file?.ContentType, readonly: true },

--- a/esp/src/src-react/components/SuperFileSummary.tsx
+++ b/esp/src/src-react/components/SuperFileSummary.tsx
@@ -1,5 +1,6 @@
 import * as React from "react";
 import { CommandBar, ContextualMenuItemType, ICommandBarItemProps, ScrollablePane, ScrollbarVisibility, Sticky, StickyPositionType } from "@fluentui/react";
+import { DFUChangeProtection, DFUChangeRestriction } from "@hpcc-js/comms";
 import { scopedLogger } from "@hpcc-js/util";
 import nlsHPCC from "src/nlsHPCC";
 import * as WsDfu from "src/WsDfu";
@@ -69,8 +70,8 @@ export const SuperFileSummary: React.FunctionComponent<SuperFileSummaryProps> = 
                 file?.update({
                     UpdateDescription: true,
                     FileDesc: description,
-                    Protect: _protected ? "1" : "2",
-                    Restrict: restricted ? "1" : "2",
+                    Protect: _protected ? DFUChangeProtection.Protect : DFUChangeProtection.Unprotect,
+                    Restrict: restricted ? DFUChangeRestriction.Restrict : DFUChangeRestriction.Unrestricted,
                 })
                     .catch(err => logger.error(err))
                     ;

--- a/esp/src/src-react/components/WorkunitSummary.tsx
+++ b/esp/src/src-react/components/WorkunitSummary.tsx
@@ -15,7 +15,6 @@ import { SlaveLogs } from "./forms/SlaveLogs";
 import { ZAPDialog } from "./forms/ZAPDialog";
 import { InfoGrid } from "./InfoGrid";
 import { WorkunitPersona } from "./controls/StateIcon";
-import { useBuildInfo } from "../hooks/platform";
 
 const logger = scopedLogger("../components/WorkunitDetails.tsx");
 
@@ -30,7 +29,6 @@ export const WorkunitSummary: React.FunctionComponent<WorkunitSummaryProps> = ({
     const [workunit, , , , refresh] = useWorkunit(wuid, true);
     const [jobname, setJobname] = React.useState("");
     const [description, setDescription] = React.useState("");
-    const [, { currencyCode }] = useBuildInfo();
     const [_protected, setProtected] = React.useState(false);
     const [showPublishForm, setShowPublishForm] = React.useState(false);
     const [showZapForm, setShowZapForm] = React.useState(false);
@@ -179,8 +177,8 @@ export const WorkunitSummary: React.FunctionComponent<WorkunitSummaryProps> = ({
                                 "owner": { label: nlsHPCC.Owner, type: "string", value: workunit?.Owner, readonly: true },
                                 "jobname": { label: nlsHPCC.JobName, type: "string", value: jobname },
                                 "description": { label: nlsHPCC.Description, type: "string", value: description },
-                                "executeCost": { label: nlsHPCC.ExecuteCost, type: "string", value: `${formatCost(workunit?.ExecuteCost ?? 0)} (${currencyCode})`, readonly: true },
-                                "fileAccessCost": { label: nlsHPCC.FileAccessCost, type: "string", value: `${formatCost(workunit?.FileAccessCost ?? 0)} (${currencyCode})`, readonly: true },
+                                "executeCost": { label: nlsHPCC.ExecuteCost, type: "string", value: `${formatCost(workunit?.ExecuteCost)}`, readonly: true },
+                                "fileAccessCost": { label: nlsHPCC.FileAccessCost, type: "string", value: `${formatCost(workunit?.FileAccessCost)}`, readonly: true },
                                 "protected": { label: nlsHPCC.Protected, type: "checkbox", value: _protected },
                                 "cluster": { label: nlsHPCC.Cluster, type: "string", value: workunit?.Cluster, readonly: true },
                                 "totalClusterTime": { label: nlsHPCC.TotalClusterTime, type: "string", value: workunit?.TotalClusterTime, readonly: true },

--- a/esp/src/src-react/components/Workunits.tsx
+++ b/esp/src/src-react/components/Workunits.tsx
@@ -9,7 +9,6 @@ import { formatCost } from "src/Session";
 import nlsHPCC from "src/nlsHPCC";
 import { useConfirm } from "../hooks/confirm";
 import { useFluentPagedGrid } from "../hooks/grid";
-import { useBuildInfo } from "../hooks/platform";
 import { HolyGrail } from "../layouts/HolyGrail";
 import { pushParams } from "../util/history";
 import { Fields } from "./forms/Fields";
@@ -81,7 +80,6 @@ export const Workunits: React.FunctionComponent<WorkunitsProps> = ({
     const [showFilter, setShowFilter] = React.useState(false);
     const [mine, setMine] = React.useState(false);
     const [uiState, setUIState] = React.useState({ ...defaultUIState });
-    const [, { currencyCode }] = useBuildInfo();
 
     //  Grid ---
     const query = React.useMemo(() => {
@@ -140,14 +138,13 @@ export const Workunits: React.FunctionComponent<WorkunitsProps> = ({
             ExecuteCost: {
                 label: nlsHPCC.ExecuteCost, width: 100,
                 formatter: function (cost, row) {
-                    return `${formatCost(cost ?? 0)} (${currencyCode || "$"})`;
+                    return `${formatCost(cost)}`;
                 }
             },
             FileAccessCost: {
                 label: nlsHPCC.FileAccessCost, width: 100,
                 formatter: function (cost, row) {
-                    return `${formatCost(cost ?? 0)
-                        } (${currencyCode || "$"})`;
+                    return `${formatCost(cost)}`;
                 }
             }
         }

--- a/esp/src/src-react/hooks/file.ts
+++ b/esp/src/src-react/hooks/file.ts
@@ -1,5 +1,5 @@
 import * as React from "react";
-import { LogicalFile, WsDfu } from "@hpcc-js/comms";
+import { LogicalFile, WsDfu, DFUDefFileFormat } from "@hpcc-js/comms";
 import { scopedLogger } from "@hpcc-js/util";
 import { singletonDebounce } from "../util/throttle";
 import { useCounter } from "./workunit";
@@ -41,7 +41,7 @@ export function useFile(cluster: string, name: string): [LogicalFile, boolean, n
     return [file, isProtected, lastUpdate, increment];
 }
 
-export function useDefFile(cluster: string, name: string, format: "def" | "xml"): [string, () => void] {
+export function useDefFile(cluster: string, name: string, format: DFUDefFileFormat): [string, () => void] {
 
     const [file] = useFile(cluster, name);
     const [defFile, setDefFile] = React.useState("");
@@ -59,10 +59,10 @@ export function useDefFile(cluster: string, name: string, format: "def" | "xml")
     return [defFile, increment];
 }
 
-export function useFileHistory(cluster: string, name: string): [WsDfu.Origin2[], () => void, () => void] {
+export function useFileHistory(cluster: string, name: string): [WsDfu.Origin[], () => void, () => void] {
 
     const [file] = useFile(cluster, name);
-    const [history, setHistory] = React.useState<WsDfu.Origin2[]>([]);
+    const [history, setHistory] = React.useState<WsDfu.Origin[]>([]);
     const [count, increment] = useCounter();
 
     const eraseHistory = React.useCallback(() => {

--- a/esp/src/src/Session.ts
+++ b/esp/src/src/Session.ts
@@ -45,7 +45,15 @@ getBuildInfo().then(info => {
     dojoConfig.currencyCode = info["currencyCode"] ?? "";
 });
 
-export const formatCost = d3Format(".2f");
+const format = d3Format(".2f");
+export function formatCost(value?: number | string): string {
+    if (value !== 0 && !value) {
+        logger.warning(`formatCost called for a nullish value: ${value}`);
+        return "";
+    }
+    const _number = typeof value === "string" ? Number(value) : value;
+    return `${format(_number)} (${dojoConfig?.currencyCode || "USD"})`;
+}
 
 export function initSession() {
     if (sessionIsActive > -1) {

